### DISCUSSION TITLE
Cast across vlan causes buffer overrun in cmd readout.

### DISCUSSION THREAD

**Discussion by ernestmusser88** [philippe44/AirConnect#393] [General]
Hello, thank you for making this program. it is really helpful. I am having a strange issue though. I have several Google Home speakers and some Raspberry pi projects on an IOT vlan with my phones and computers on a seperate home network. AirConnect is running on the IOT network and I am able to see my devices and begin casting to them. I get volume and transport control on the Google Home speaker, but when I hit play the command line begins displaying bufferoverruns and I get no audio. I have put specific traffic rules in place for known ports and ive tried forwarding all ports between the network. The system works without issues if the phone is connected to the IOT network. Could you provide any advice for further troubleshooting? Thank you.

**Comment by philippe44** [comment]
There is an HTTP server opened for the CC player to grab the audio. If you don't hear anything and have a structured lan construct, it might be that it is unreachable. Other then that, I'd need to see the log

**Reply by ernestmusser88** [reply]
Understood, I'll get some logs for you. If my firewall rules were blocking the webserver, would that break casting from web browsers and my Pixel phone? Thank you.

**Reply by philippe44** [reply]
probably not

**Reply by ernestmusser88** [reply]
[AirConnect_PrintOut.txt](https://github.com/philippe44/AirConnect/files/7859571/AirConnect_PrintOut.txt)

Thank you for taking a look at this.

**Reply by philippe44** [reply]
hmmm... so AirCast is on the 50.x branch, always right? This is the IOT branch, I assume. What is weird is that when you don't have audio, AirCast received the HTTP GET, responds to it but the CC player never grabs anything while the iPhone keeps sending audio, hence we end in buffer overflow. In the other case, everything works the same but the CC devices grabs audio correctly... That does not make sense :-(

I would have expected that the iPhone cannot send audio and you just have silence on CC, but it's not what seems to happen. At this point, you should try some wireshark checks to see what's happening on this TCP session use the HTTP GET request. Unless I misunderstood something about your setup.

**Reply by ernestmusser88** [reply]
You are correct. The iPad is on the 1.x network while the AirCast and the CC Players are on the 50.x network. I initally had Aircast on the 1.x network, but I encountered this issue so I moved it to the IOT lan to see if anything changed. I'll set up a mirroring port and do some pcaps when I have some more time. I'll let you know what I find out. Thank you for your time thus far.

**Reply by Jpereyra316** [reply]
I'm in a VERY similar situation experiencing the same issue. I have setup VLANs for LAN, IoT, and Guest. I want any device on any network to be able to cast. However, I can only cast when in the IoT VLAN.

**Reply by Jpereyra316** [reply]
Here is a packet capture of what it looks like when attempting to cast over separate VLANs.
Rename attachment from .log to .cap.

[AirConnect_Capture_Bad_cap.log](https://github.com/philippe44/AirConnect/files/10316599/AirConnect_Capture_Bad_cap.log)

**Reply by skitchbeatz** [reply]
@Jpereyra316 Are you still having this issue? I'm unable to cast to aircast targets for the same reason

**Reply by Jpereyra316** [reply]
Last time I tested, yes. I've been casting from the same network lately.

**Comment by ernestmusser88** [comment]
I'm planning to dig deeper into this when I have more time. If I figure it
out, I'll setup a fork and let you know.

Best

Ernest
